src/sdk/pynni/nni/smac_tuner/convert_ss_to_scenario.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge,
+# to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction,
+# including without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import os
+import json
+
+def get_json_content(file_path):
+    '''Load json file content'''
+    try:
+        with open(file_path, 'r') as file:
+            return json.load(file)
+    except TypeError as err:
+        print('Error: ', err)
+        return None
+
+def generate_pcs(nni_search_space_content):
+    # parameter_name categorical {value_1, ..., value_N} [default value]
+    # parameter_name ordinal {value_1, ..., value_N} [default value]
+    # parameter_name integer [min_value, max_value] [default value]
+    # parameter_name integer [min_value, max_value] [default value] log
+    # parameter_name real [min_value, max_value] [default value]
+    # parameter_name real [min_value, max_value] [default value] log
+    # https://automl.github.io/SMAC3/stable/options.html
+    search_space = nni_search_space_content
+    with open('param_config_space.pcs', 'w') as pcs_fd:
+        if isinstance(search_space, dict):
+            for key in search_space.keys():
+                if isinstance(search_space[key], dict):
+                    try:
+                        if search_space[key]['_type'] == 'choice':
+                            pcs_fd.write('%s categorical {%s} [%s]\n' % (
```

Comment:
can you try to split this different condition into different function?  too long...too complex here...

Reply:
done